server/cluster/unsafe_recovery_controller.go

```diff
@@ -79,6 +80,18 @@ func (u *unsafeRecoveryController) RemoveFailedStores(failedStores map[uint64]st
 		return errors.Errorf("Another request is working in progress")
 	}
 	u.reset()
+	for failedStore := range failedStores {
+		store := u.cluster.GetStore(failedStore)
+		if store != nil && store.IsUp() && !store.IsDisconnected() {
+			return errors.Errorf("Store %v is up and connected", failedStore)
+		}
+	}
+	for failedStore := range failedStores {
```

Comment:
can we do it in the above loop?

Reply:
They are intentionally separated since we want to bury stores if and only if all of them are able to be buried.